Zend/zend_compile.c

```diff
@@ -837,7 +927,7 @@ uint32_t zend_add_member_modifier(uint32_t flags, uint32_t new_flag) /* {{{ */
 			"Multiple readonly modifiers are not allowed", 0);
 		return 0;
 	}
-	if ((new_flags & ZEND_ACC_ABSTRACT) && (new_flags & ZEND_ACC_FINAL)) {
+	if (target == ZEND_MODIFIER_TARGET_METHOD && (new_flags & ZEND_ACC_ABSTRACT) && (new_flags & ZEND_ACC_FINAL)) {
 		zend_throw_exception(zend_ce_compile_error,
 			"Cannot use the final modifier on an abstract class member", 0);
```

Comment:
It would be nice to clarify the target in the message (method or property). As far as I remember, we rarely mention `member` in error messages. As far as I quickly saw, there was a `Call to a member function %s() on %s` exception, but this sounds very awkward

Reply:
`static` is  not valid for properties, so we should just adjust the message to say `method` here. Properties will be caught earlier now.